modules/exploits/linux/http/progress_flowmon_unauth_cmd_injection.rb

```diff
@@ -76,27 +75,20 @@ def check
       'method' => 'GET'
     )
 
-    unless res
-      print_error('Connection failed')
-      return CheckCode::Unknown
-    end
+    return CheckCode::Unknown('Connection failed') unless res
+    return CheckCode::Safe('Target does not appear to be running Progress Flowmon') unless res.code == 200 && res.get_xml_document.xpath('//title').text == 'Flowmon Web Interface'
```

Comment:
Is the response an HTML page? If so, maybe `#get_html_document` should be used instead. This methods returns a `Nokogiri::HTML` object, which seems to be more adequate here.
```suggestion
    return CheckCode::Safe('Target does not appear to be running Progress Flowmon') unless res.code == 200 && res.get_html_document.xpath('//title').text == 'Flowmon Web Interface'
```

Reply:
It is an html document, great catch! That was a copy pasta error on my part.